### ISSUE TITLE
Add support for printing violations in a external file

### ISSUE THREAD

**Issue by abhinaba-ghosh** [abhinaba-ghosh/axe-playwright#12]
It may be a good option for the users to have a detailed HTML report so that the artifact can be shared across.

**Comment by BeataKr** [comment]
To get html report you can easily use npm axe-html-reporter pack, example of usage:

const violations = await getViolations(...) // from axe-playwright

if (violations.length > 0) {
          createHtmlReport({ // from axe-html-reporter
            results: {
              violations: violations,
            },
            options: {
              outputDir: resultDir,
              reportFileName: `${this.testName}.html`,
            },
          });
        }

The author can use the same solution or use his own HTML parser.

**Comment by abhinaba-ghosh** [comment]
@BeataKr Thanks very useful. Would you mind integrate it with the current utility internally? Would appreciate a PR

**Comment by BeataKr** [comment]
Yes, when I'll find a time for this, with pleasure :D

**Comment by BeataKr** [comment]
@abhinaba-ghosh can you have a look on https://github.com/abhinaba-ghosh/axe-playwright/pull/41 

Example (working) usage:
```
 const axeAudit = async auditName => {
    await getAndSaveViolations(page, undefined, {
      axeOptions: {
        runOnly: {
          type: 'tag',
          values: ['wcag2a', 'wcag2aa', 'wcag21aa']
        }
      }
    },
      {
        outputDirPath: RESULTS_DIR,
        outputDir: 'accessibility',
        reportFileName: 'accessibility-audit-' +auditName.replace(/\s/g, '-') + '.html'
      })
  };
```

**Comment by Zohairtoo** [comment]
Hi,
I have started using Playwright with Axe on TypeScript. Currently I can print Reports of Axe violations through console.log. But I want to use 'axe-html-reporter'. But When Im trying it is giving me error. Im using the following code:
------------------------------------------------------------------------------------------------------------
import { Page, expect } from '@playwright/test'
import {injectAxe, checkA11y, getViolations, reportViolations} from 'axe-playwright'
import { createHtmlReport } from 'axe-html-reporter';
import {test} from "@playwright/test"

test('exampleTest01 with Axe Injected in it', async ({page}) => {
    await page.goto('https://apps.esignlive.com/login');

    const violations = await getViolations(page) // from axe-playwright

if (violations.length > 0) {
          createHtmlReport({ // from axe-html-reporter
            results: {
              violations: violations,
            },
            options: {
              outputDir: 'resultDir',
              reportFileName: 'fileName.html',
            },
          });
        }
    });
-------------------------------------------------------------------------------------------------------
but I get the following error:

page.evaluate: TypeError: Cannot read properties of undefined (reading 'run')

       9 |     await page.goto('https://apps.esignlive.com/login');
      10 |
    > 11 |     const violations = await getViolations(page) // from axe-playwright
         |                              ^
      12 |
      13 | if (violations.length > 0) {
      14 |           createHtmlReport({ // from axe-html-reporter
-------------------------------------------------------------------------------------------------------

Do I need to initialize page and if yes then how?

**Comment by abhinaba-ghosh** [comment]
I am checking this today 💪

**Comment by Zohairtoo** [comment]
Hello Abhinaba,

Did you happen to find any solution to it?

**Comment by abhinaba-ghosh** [comment]
Apologies for the delay. I am really busy with some critical releases. Making a room for the next weekend for sure.

**Comment by Zohairtoo** [comment]
Hi Abhinaba,

I managed to solve with the following code 
test('exampleTest01 with Axe Injected in it', async ({ page }) => {
    await page.goto('webSiteAddressURL');

    await injectAxe(page);
    const violations = await getViolations(page) // from axe-playwright

    if (violations.length > 0) {
        createHtmlReport({ // from axe-html-reporter
            results: {
                violations: violations,
            },
            options: {
                outputDir: 'resultDir',
                reportFileName: 'fileName.html',
            },
        });
    }
});

Actually there was a dependency I was using older version and this"[^1.18.1]" is the correct version.

**Comment by abhinaba-ghosh** [comment]
Hi @Zohairtoo that's great. Would you mind raising a PR to integrate this behaviour in the library itself?

**Comment by BeataKr** [comment]
Due to https://github.com/abhinaba-ghosh/axe-playwright/pull/209 released with 1.2.1 we can close, right?
